DataManagementSystem/DB/FileCatalogDB.py

```diff
@@ -262,62 +262,118 @@ def hasAccess( self, opType, paths, credDict ):
   #  Path based read methods
   #
 
-  def changePathOwner( self, lfns, credDict, recursive = False ):
-    """ Change the owner of the given list of paths
-    """
-    res = self._checkPathPermissions( 'changePathOwner', lfns, credDict )
-    if not res['OK']:
-      return res
-    failed = res['Value']['Failed']
-    successful = dict()
-    if res['Value']['Successful']:
-      res = self.fileManager.changePathOwner( res['Value']['Successful'], credDict, recursive )
-      if not res['OK']:
-        return res
-      failed.update( res['Value']['Failed'] )
-      successful = res['Value']['Successful']
+  def changePathOwner( self, paths, credDict, recursive = False ):
+    """ Bulk method to change Owner for the given paths
 
-    return S_OK({'Successful':successful,'Failed':failed}) 
-  
-  def changePathGroup(self, lfns, credDict, recursive=False):
-    """ Change the group of the given list of paths
+        :param dict paths: dictionary < lfn : owner >
+        :param dict credDict: dictionary of the caller credentials
+        :param boolean recursive: flag to apply the operation recursively
     """
-    res = self._checkPathPermissions( 'changePathGroup', lfns, credDict )
+    res = self._checkPathPermissions( 'changePathOwner', paths, credDict )
     if not res['OK']:
       return res
     failed = res['Value']['Failed']
+    result = self.__changePathFunction( res['Value']['Successful'], credDict,
+                                        self.dtree.changeDirectoryOwner,
+                                        self.fileManager.changeFileOwner,
+                                        recursive = recursive )
+    failed.update( result['Value']['Failed'] )
+    successful = result['Value']['Successful']
+    return S_OK( { 'Successful':successful, 'Failed':failed } )
 
-    # if no successful, just return
-    if not res['Value']['Successful']:
-      return S_OK( {'Successful':{}, 'Failed':failed} )
+  def changePathGroup( self, paths, credDict, recursive = False ):
+    """ Bulk method to change Group for the given paths
 
-    res = self.fileManager.changePathGroup( res['Value']['Successful'], credDict, recursive )
+        :param dict paths: dictionary < lfn : group >
+        :param dict credDict: dictionary of the caller credentials
+        :param boolean recursive: flag to apply the operation recursively
+    """
+    res = self._checkPathPermissions( 'changePathGroup', paths, credDict )
     if not res['OK']:
       return res
-    failed.update( res['Value']['Failed'] )
-    successful = res['Value']['Successful']
+    failed = res['Value']['Failed']
+    result = self.__changePathFunction( res['Value']['Successful'], credDict,
```

Comment:
You must test here and in the others that res['Value']['Successful'] is not empty to avoid the "ERROR:empty input[]" problem that I fixed in a previous PR